src/query/procedure/module.cpp

```diff
@@ -832,10 +833,30 @@ PythonModule::~PythonModule() {
 }
 
 bool PythonModule::Load(const std::filesystem::path &file_path) {
+  // This configures python to load share libs with RTLD_NOW and RTLD_DEEPBIND
+  // flags, which in turn solves the issue of loading wrong libstdc++ lib.
+  static bool is_python_initialized = false;
+  if (!is_python_initialized) [[unlikely]] {
+    auto gil = py::EnsureGIL();
+    // NOLINTNEXTLINE(hicpp-signed-bitwise)
+    auto *flag = PyLong_FromLong(RTLD_NOW | RTLD_DEEPBIND);
+    auto *setdl = PySys_GetObject("setdlopenflags");
+    MG_ASSERT(setdl);
+    auto *arg = PyTuple_New(1);
+    MG_ASSERT(arg);
+    MG_ASSERT(PyTuple_SetItem(arg, 0, flag) == 0);
+    PyObject_CallObject(setdl, arg);
+    Py_DECREF(flag);
+    Py_DECREF(setdl);
+    Py_DECREF(arg);
+    is_python_initialized = true;
+  }
```

Comment:
We have a Python initialization related code in [memgraph.cpp](https://github.com/memgraph/memgraph/blob/bf01c58ed9950a1b2d05a884aad9544e8193b7f3/src/memgraph.cpp#L1050-L1059): https://github.com/memgraph/memgraph/blob/bf01c58ed9950a1b2d05a884aad9544e8193b7f3/src/memgraph.cpp#L1050-L1059

I think this and the existing python initialization should be in one place. Maybe you can move this to `main`?